modules/media_riddle_marketplace/src/RiddleMediaService.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Drupal\media_riddle_marketplace;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\riddle_marketplace\RiddleFeedServiceInterface;
+
+/**
+ * Class RiddleFeedService.
+ *
+ * @package Drupal\riddle_marketplace
+ */
+class RiddleMediaService implements RiddleMediaServiceInterface {
+
+  /**
+   * The riddle feed service.
+   *
+   * @var \Drupal\riddle_marketplace\RiddleFeedServiceInterface
+   */
+  protected $feedService;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Riddle Media Service.
+   *
+   * Constructor.
+   *
+   * @param \Drupal\riddle_marketplace\RiddleFeedServiceInterface $feedService
+   *   Riddle Feed service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   */
+  public function __construct(RiddleFeedServiceInterface $feedService, EntityTypeManagerInterface $entityTypeManager) {
+    $this->feedService = $feedService;
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function createMediaEntities() {
+
+    foreach ($this->getNewRiddles() as $bundle => $riddles) {
+      /** @var \Drupal\media_entity\MediaBundleInterface $bundle */
+      $bundle = $this->entityTypeManager->getStorage('media_bundle')
+        ->load($bundle);
+      $sourceField = $bundle->getTypeConfiguration()['source_field'];
+
+      foreach ($riddles as $riddle) {
+        $this->entityTypeManager->getStorage('media')->create([
+          'bundle' => $bundle->id(),
+          $sourceField => $riddle,
+        ])->save();
+      }
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getNewRiddles() {
+
+    $feed = $this->feedService->getFeed();
+
+    $riddleIds = array_column($feed, 'uid');
```

Comment:
find a better naming to make riddles from the feed and riddles from the storage easier to distinguish